### PR TITLE
matc: add --raw argument for internal use.

### PR DESCRIPTION
**Pull request by prideout**
This feature can generate SPIRV blobs and will be used by VulkanBlitter.

(#3485)

### CODE REVIEW

File: tools/matc/src/matc/MaterialCompiler.cpp [hunk 1]
```diff
@@ -441,4 +453,49 @@ bool MaterialCompiler::parseMaterial(const char* buffer, size_t size,
...
+    msg = (EShMessages) (EShMessages::EShMsgVulkanRules | EShMessages::EShMsgSpvRules);
+
+    tShader.setAutoMapBindings(true);
+
+    bool ok = tShader.parse(&DefaultTBuiltInResource, version, false, msg);
+    if (!ok) {
+        std::cerr << "ERROR: Unable to parse " << ext << ":" << std::endl;
+        std::cerr << tShader.getInfoLog() << std::endl;
+        return false;
+    }
```
Comment (bejado) [inline tools/matc/src/matc/MaterialCompiler.cpp:479]: I think we need to make sure we link the program, just as we do for normal materials:
https://github.com/google/filament/blob/9f68daeb8f18b8d13e66105a009e59644531a8a2/libs/filamat/src/GLSLPostProcessor.cpp#L202-L209

**Review by bejado — CHANGES_REQUESTED** [review]

File: tools/matc/src/matc/MaterialCompiler.cpp (hunk 1 above)
Comment (prideout) [reply tools/matc/src/matc/MaterialCompiler.cpp:478]: Good catch, fixed.

**Comment by bejado** [comment]
Hm, Windows fails with

```
CUSTOMBUILD : error : Unable to parse vs: [D:\a\filament\filament\out\cmake-mt\filament\backend\vkshaders.vcxproj]
CUSTOMBUILD : error : 0:12: '' :  syntax error, unexpected DOT [D:\a\filament\filament\out\cmake-mt\filament\backend\vkshaders.vcxproj]
CUSTOMBUILD : error : 1 compilation errors.  No code generated. [D:\a\filament\filament\out\cmake-mt\filament\backend\vkshaders.vcxproj]
```

I can't think of why though..

**Comment by prideout** [comment]
> Hm, Windows fails

I have a theory that this is due to a lack of null terminator.  Amended a commit, fingers crossed.

**Pull request merged**